sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/generator/NativeQueryGenerator.java

```diff
@@ -40,8 +42,22 @@ public SqlQuerySpec generateSortedQuery(SqlQuerySpec querySpec, Sort sort) {
         if (sort == null || sort.isUnsorted()) {
             return querySpec;
         } else {
-            String querySort = AbstractQueryGenerator.generateQuerySort(sort);
-            String queryText = "select * from (" + querySpec.getQueryText() + ") r " + querySort;
+            Pattern pattern1 = Pattern.compile("\\s(?i)from root\\s");
+            Pattern pattern2 = Pattern.compile("\\s(?i)from\\s");
+            Matcher matcher = pattern1.matcher(querySpec.getQueryText());
+            int beginIndex = 0;
+            if (matcher.find()) {
+                beginIndex = matcher.start(0) + 11;
+            } else {
+                matcher = pattern2.matcher(querySpec.getQueryText());
+                matcher.find();
+                beginIndex = matcher.start(0) + 6;
+            }
+            String tableName = querySpec.getQueryText().substring(beginIndex);
+            tableName = tableName.substring(0, tableName.indexOf(" "));
+
+            String querySort = AbstractQueryGenerator.generateQuerySort(sort, tableName);
+            String queryText = querySpec.getQueryText() + " " + querySort;
```

Comment:
@kushagraThapar I am not sure how I feel about this solution, do you think it is to complex with the two regex checks? This function is public so if a user accesses it directly it will be a "Select * from root r..." query, however our internal code never calls this function with a root query like that. Our internal code only calls this function with @query annotation, although I suppose a user could define that as a "Select * from root r..." query also if they choose to.

Reply:
I wouldn't worry about user using it directly. Given the class description already says that this is only for `@query annotated` queries
/**
 * Augment custom queries sourced from @Query annotations
 */

If user wants to use it, I would let it fail on its face, rather complicating the logic here. 

I feel we should simplify this code, and only rely on queries coming from `@query annotated` queries. We don't want customers to use this class. 
Adding more warning comments to both of these APIs `generateSortedQuery()` and `generateCountQuery()` should help with clarification on their purposes. 
Also, please add the same warning comment to the class itself. 

Regarding your regex, we should add comments on what kind of queries we are expecting here, and what we are doing with regex.

Reply:
Done.

Reply:
Done.